src/VisualStudio/Core/Def/Implementation/TableDataSource/AbstractRoslynTableDataSource.cs

```diff
@@ -40,26 +49,23 @@ private void ConnectToSolutionCrawlerService(Workspace workspace)
             }
 
             var reporter = crawlerService.GetProgressReporter(workspace);
+            reporter.ProgressChanged += OnSolutionCrawlerProgressChanged;
 
             // set initial value
-            IsStable = !reporter.InProgress;
-
-            ChangeStableState(stable: IsStable);
-
-            reporter.Started += OnSolutionCrawlerStarted;
-            reporter.Stopped += OnSolutionCrawlerStopped;
+            SolutionCrawlerProgressChanged(reporter.InProgress);
         }
 
-        private void OnSolutionCrawlerStarted(object sender, EventArgs e)
+        private void OnSolutionCrawlerProgressChanged(object sender, bool running)
         {
-            IsStable = false;
-            ChangeStableState(IsStable);
+            SolutionCrawlerProgressChanged(running);
         }
 
-        private void OnSolutionCrawlerStopped(object sender, EventArgs e)
+        private void SolutionCrawlerProgressChanged(bool running)
         {
-            IsStable = true;
+            IsStable = !running;
             ChangeStableState(IsStable);
+
+            _reporterOpt?.Started(running);
```

Comment:
It seems super weird that the table is responsible for notifying the reporter.

Reply:
not sure why. I dont want create unnecessary indirection making data source to raise events which reporter to listen. if there is any other listener than progress reporter than yes. but until then, I dont think we need that indirection.